push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/PushyApnsSender.java

```diff
@@ -208,35 +212,42 @@ private ApnsClient buildApnsClient(final iOSVariant iOSVariant) {
         // this check should not be needed, but you never know:
         if (iOSVariant.getCertificate() != null && iOSVariant.getPassphrase() != null) {
 
-            // add the certificate:
-            try {
-                final ByteArrayInputStream stream = new ByteArrayInputStream(iOSVariant.getCertificate());
-
-                final ApnsClientBuilder builder = new ApnsClientBuilder();
-                builder.setClientCredentials(stream, iOSVariant.getPassphrase());
-
-                if (ProxyConfiguration.hasHttpProxyConfig()) {
-                    if (ProxyConfiguration.hasBasicAuth()) {
-                        String user =  ProxyConfiguration.getProxyUser();
-                        String pass = ProxyConfiguration.getProxyPass();
-                        builder.setProxyHandlerFactory(new HttpProxyHandlerFactory(ProxyConfiguration.proxyAddress(), user, pass));
-                    } else {
-                        builder.setProxyHandlerFactory(new HttpProxyHandlerFactory(ProxyConfiguration.proxyAddress()));
-                    }
+            // only process valid certs!
+            if (ApnsUtil.checkValidity(iOSVariant.getCertificate(), iOSVariant.getPassphrase().toCharArray())) {
 
-                } else if (ProxyConfiguration.hasSocksProxyConfig()) {
-                    builder.setProxyHandlerFactory(new Socks5ProxyHandlerFactory(ProxyConfiguration.socks()));
-                }
+                // add the certificate:
+                try {
+                    final ByteArrayInputStream stream = new ByteArrayInputStream(iOSVariant.getCertificate());
```

Comment:
we (correctly) close this stream even if it is a ByteArrayInputStream. Would be better to change this to a "try with resource"